### PR TITLE
feat: ampl distribution

### PR DESCRIPTION
**Pull request by sakulstra**

### CODE REVIEW

File: src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/InterimAAMPLDistribution.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+With [aip 72](https://vote.onaave.com/proposal/?proposalId=72&ipfsHash=0xaa46d2cf629d68cc84bcc83156b2fd8e54819c5e848c229c7e62d1f6212886cc) having passed the governance process, aAMPL transfers are no longer permitted, which allows to snapshot the current aAMPL balances to perform a fair distribution between affected users.
+
+## Specification
+
+The proposal will perform the following steps upon execution:
+
+- withdraw 301.5k USDC from the collector (300k for the distribution and 1.5k as fee for angle labs)
+- approve the full amount to [0x8BB4C975Ff3c250e0ceEA271728547f3802B36Fd](https://etherscan.io/address/0x8BB4C975Ff3c250e0ceEA271728547f3802B36Fd) which is the distribution creator by angle labs
+- sign the tos of https://app.merkl.xyz/
```
Comment (eboadom) [inline src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/InterimAAMPLDistribution.md:26]:
```suggestion
- sign the ToS of https://app.merkl.xyz/, a requirement on the Merkl platform
```

**Review by eboadom — APPROVED** [review]

File: src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/AaveV2Ethereum_InterimAAMPLDistribution_20240409.sol [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    AaveV2Ethereum.POOL.withdraw(REWARD_TOKEN, REWARD_AMOUNT_PLUS_FEES, address(this));
+    // 2. approve to merkl
+    IERC20(REWARD_TOKEN).approve(address(DISTRIBUTION_CREATOR), REWARD_AMOUNT_PLUS_FEES);
+    // 3. accept tos
+    DISTRIBUTION_CREATOR.acceptConditions();
+    // 4.create campaign
+    IDistributionCreator.CampaignParameters memory newCampaign = IDistributionCreator
+      .CampaignParameters({
+        campaignId: '',
+        creator: address(0),
```
Comment (sendra) [inline src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/AaveV2Ethereum_InterimAAMPLDistribution_20240409.sol:73]: should this be executor? or is it correct to leave it at 0?

File: src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/AaveV2Ethereum_InterimAAMPLDistribution_20240409.sol [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        rewardToken: REWARD_TOKEN,
+        amount: REWARD_AMOUNT_PLUS_FEES,
+        campaignType: 4,
+        startTimestamp: uint32(block.timestamp + 2 hours),
+        duration: 1 hours,
```
Comment (sendra) [inline src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/AaveV2Ethereum_InterimAAMPLDistribution_20240409.sol:78]: isn't 1 hour too little?

File: src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/InterimAAMPLDistribution.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+Therefore the proposal will perform the following steps upon execution:
+
+- withdraw USDC from the collector (298.5k including a 0.5% fee for angle labs)
+- approve the full amount to [0x8BB4C975Ff3c250e0ceEA271728547f3802B36Fd](https://etherscan.io/address/0x8BB4C975Ff3c250e0ceEA271728547f3802B36Fd) which is the distribution creator by Angle Labs
+- sign the tos of https://app.merkl.xyz/ via a onchain transaction, a requirement on the Merkl platform
+- create a campaign to distribute funds to the affected users
+
+2 hours after proposal execution, users will be able to claim the USDC on https://app.merkl.xyz/
+
```
Comment (sendra) [inline src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/InterimAAMPLDistribution.md:44]: If for what i understood the campaign is only for 1 hour, it should be specified here

File: src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/InterimAAMPLDistribution.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+- Only claims over value of 30 USDC have been included, given that gas-wise, would not be profitable to claim lower amounts. However, those values lower than 30 USDC will be naturally taken into account for the final follow-up distribution.
+- Once again, _this is an interim distribution, that will be follow up by another with more a precise numbers and bigger in size_.
+- For the sake of reducing complexity, the Aave governance proposal will release the whole 300'000 USDC, and the Ampleforth team can just transfer to the Aave Collector the 40% of that amount, removing any dependency for users to claim as soon as possible.
+- It is possible to check each address claims [HERE](https://github.com/bgd-labs/aave-proposals-v3/blob/8d5b3e902adf7c5c246e752b5d6b6e0e5d9831b7/src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/distribution.pdf).
```
Comment (sendra) [inline src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/InterimAAMPLDistribution.md:28]: wouldnt it be better to link the ipfs file? https://angle-blog.infura-ipfs.io/ipfs/Qmb9uJbEdppQsL8W4aVKxREoHo42iXtcp4CV1FLE5tY8Rt ?

File: src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/AaveV2Ethereum_InterimAAMPLDistribution_20240409.sol (hunk 2 above)
Comment (sakulstra) [reply src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/AaveV2Ethereum_InterimAAMPLDistribution_20240409.sol:73]: It defaults to msg.sender and it's what angle recommended to put

File: src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/AaveV2Ethereum_InterimAAMPLDistribution_20240409.sol (hunk 3 above)
Comment (sakulstra) [reply src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/AaveV2Ethereum_InterimAAMPLDistribution_20240409.sol:78]: This 1hr is arbitrary. For airdrops is not used, but contract side there is validation to be non zero and multiple of 3600

Comment (eboadom) [reply src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/AaveV2Ethereum_InterimAAMPLDistribution_20240409.sol:78]: Lukas can comment on this, but it doesn't mean that after 1 hour funds will not be claimable

File: src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/InterimAAMPLDistribution.md (hunk 4 above)
Comment (eboadom) [reply src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/InterimAAMPLDistribution.md:44]: Previous comment

Comment (sakulstra) [reply src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/InterimAAMPLDistribution.md:44]: It's not the duration param is arbitrary but must be set due to the generic interface

File: src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/InterimAAMPLDistribution.md (hunk 5 above)
Comment (sakulstra) [reply src/20240409_AaveV2Ethereum_InterimAAMPLDistribution/InterimAAMPLDistribution.md:28]: Ernesto said to better put sth for formatted numbers

**Pull request merged**